tests/test_gradient_boosting.py

```diff
@@ -192,6 +192,12 @@ def custom_check_estimator(Estimator):
             # X is both Fortran and C aligned and numba can't compile.
             # Opened numba issue 3569 (not sure if this is by design)
             continue
+        if check is estimator_checks.check_classifiers_train:
+            continue # probas don't exactly sum to 1 (very close though)
```

Comment:
wouldn't it be possible to post process the softmax to renormalize instead?

Reply:
I guess... but that's some extra computation just for the test to pass.

The sum of probas are really close to 1:

Changing `assert np.sum(y_prob, axis=1), np.ones(n_samples))` to `assert_array_almost_equal(np.sum(y_prob, axis=1), np.ones(n_samples))` 
make the check pass.

Alternatively we could switch to float64...

Reply:
Alright. Maybe we could think of relaxing this check in scikit-learn when `predict_proba` has 32-bit dtype. But this is fine for now.